http-client/src/http_client.rs

```diff
@@ -52,29 +49,28 @@ fn format_headers(h: &HeaderMap) -> HashMap<String, String> {
 
 fn build_request(
     client: &Client,
-    req: Request,
-) -> Result<RequestBuilder, Box<dyn std::error::Error>> {
-    let path = req.path.as_str();
+    req: RequestArgs,
+) -> Result<RequestBuilder, Box<dyn std::error::Error + Send + Sync>> {
     let mut r = match req.method.as_str() {
         "GET" => {
-            let r = client.get(format!("{}/{}", path, req.query_string).as_str());
+            let r = client.get(&req.url);
             Ok(r)
         }
         "POST" => {
-            let r = client.post(path).body(req.body);
+            let r = client.post(&req.url).body(req.body);
             Ok(r)
         }
-        "HEAD" => Ok(client.head(path)),
-        "PUT" => Ok(client.put(path)),
-        "DELETE" => Ok(client.delete(path)),
-        "PATCH" => Ok(client.patch(path)),
-        "OPTIONS" => Ok(client.request(reqwest::Method::OPTIONS, path)),
-        "CONNECT" => Ok(client.request(reqwest::Method::CONNECT, path)),
-        "TRACE" => Ok(client.request(reqwest::Method::TRACE, path)),
+        "HEAD" => Ok(client.head(&req.url)),
+        "PUT" => Ok(client.put(&req.url)),
```

Comment:
The put doesn't appear to submit a body. `PUT` usually requires content. I know this appears to have been broken in the original version of this provider, too.